spring-integration-core/src/main/java/org/springframework/integration/splitter/AbstractMessageSplitter.java

```diff
@@ -145,4 +172,81 @@ public String getComponentType() {
 	 */
 	protected abstract Object splitMessage(Message<?> message);
 
+	/**
+	 * A {@link HashMap} implementation that allows adding headers but disallows removal.
+	 *
+	 * @since 4.2
+	 *
+	 */
+	@SuppressWarnings("serial")
+	protected class PutOnlyHeaders extends HashMap<String, Object> {
+
+		public PutOnlyHeaders(Map<? extends String, ? extends Object> m) {
+			super(m);
```

Comment:
I thought you convinced me yesterday that we can trust target implementations, hence this new abstraction with copy ctor is an impact for the performance.